src/Controller/SecurityController.php

```diff
@@ -33,4 +42,49 @@ public function logout()
     {
         throw new \LogicException('This method can be blank - it will be intercepted by the logout key on your firewall.');
     }
+
+    /**
+     * @Route("/admin/user/{id}/set-password", name="admin_user_set_password")
+     * @IsGranted("ROLE_ADMIN")
+     */
+    public function setUserPassword(Request $request, User $user, TranslatorInterface $translator, UserManager $userManager, EntityManagerInterface $entityManager)
+    {
+        $minLength = SetUserPasswordFormType::MIN_LENGTH;
+
+        $form = $this->createForm(SetUserPasswordFormType::class, [
+            'password' => ByteString::fromRandom(2 * $minLength),
+        ]);
+
+        $form->handleRequest($request);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $password = trim($form->getData()['password'] ?? '');
+            if (strlen($password) < $minLength) {
+                $this->addFlash('danger', $translator->trans('User password too short. Must be contain least %min_length% characters.', [
+                    '%min_length%' => $minLength,
+                ]));
+            } else {
```

Comment:
Wouldn't `$form->isValid()` already have caught this? Or is this due to the `trim` method call above?

Reply:
Yes. No.